### ISSUE TITLE
Request for basic encryption example

### ISSUE THREAD

**Issue by VR-Architect** [Unity-Technologies/Addressables-Sample#40]
Please provide a basic example of implementing some basic level of encryption. Thanks.

**Comment by ChetRippo** [comment]
i agree, this would be very useful to have. Would like to not have to worry about assets being flipped right after launch.

**Comment by renatoi** [comment]
In multiple Unite talks the dev mentioned that an encryption example would be available on Github. Yet, I couldn't find them. Has this been abandoned?

**Comment by kirstenpilla** [comment]
Hi all apologies for the delay! Unfortunately there are no plans to provide an example project demonstrating encryption. There are some limitations such as bundle caching and memory increases that make it not feasible, especially for projects with lots of AssetBundles.

**Comment by thomasftg** [comment]
> Hi all apologies for the delay! Unfortunately there are no plans to provide an example project demonstrating encryption. There are some limitations such as bundle caching and memory increases that make it not feasible, especially for projects with lots of AssetBundles.

Can you elaborate a bit more so we can find a solution?   Specifically is it NOT possible or does the team not have the bandwidth to provide the sample given some quick prototyping issues being encountered (caching/mem increase you mention).

**Comment by kirstenpilla** [comment]
@thomasftg Hi, yes the latter. It is possible to do for projects with a small number of AssetBundles. Due the issues mentioned we are not able to provide an official sample project at this time. 

You could create a custom AssetBundleProvider class that extends that base class, and additionally handles the encryption and decryption steps. One way to approach this is to use LZ4 Decryption. An AssetBundle that uses LZ4 compression consists of 256KB chunks that can be encrypted. The bundle can be loaded from a seekable stream (AssetBundle.LoadFromStreamAsync) and individual chunks can be decrypted as they are loaded.
